packaging/technical-addon/packaging-scripts/cicd-tests/happypath-test.sh

```diff
@@ -0,0 +1,108 @@
+while [ $ATTEMPT -le $MAX_ATTEMPTS ]; do
+    curl --header "Content-Type:application/json" --header "X-SF-TOKEN:${OLLY_ACCESS_TOKEN}" "https://api.us0.signalfx.com/v2/metrictimeseries?query=host.name:${otel_hostname}%20AND%20sf_metric:otelcol_process_uptime%20AND%20splunk.distribution:otel-ta" > "$TEST_FOLDER/uptime.json"
+    count=$( grep '"count"' "$TEST_FOLDER/uptime.json" | awk -F ':\ ' '{print $2}' | awk -F ',' '{print $1}')
+
+    if [[ "$count" -gt "0" ]] && jq '[.results[].created, .results[].lastUpdated] | max as $max | $max >= ($ENV.CUTOFF | tonumber)' "$TEST_FOLDER/uptime.json" ; then
+        break
+    else
+        ATTEMPT=$((ATTEMPT + 1))
+        sleep $DELAY
+    fi
+done
+if [ $ATTEMPT -gt $MAX_ATTEMPTS ]; then
+    echo "Failed find metrics in last 5m after $MAX_ATTEMPTS attempts."
+    cat "$GATEWAY_AGENT_LOGS_DIR/otel.log"
+    exit 1
+fi
+
+# Verify the addon can be restarted successfully
+orca_container_name=$(splunk_orca --cloud ${ORCA_CLOUD} --printer json show --deployment-id "${deployment_id}" containers |  jq -r '.[keys[0]] | .[keys[0]] | .containers | keys[0]')
+splunk_orca --cloud "${ORCA_CLOUD}" exec --exec-user splunk "${orca_container_name}" '/opt/splunk/bin/splunk restart'
+sleep 90s
+
+MAX_ATTEMPTS=30
+DELAY=10
+ATTEMPT=1
+while [ $ATTEMPT -le $MAX_ATTEMPTS ]; do
+    scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -r -i ~/.orca/id_rsa "splunk@$ip_addr:/opt/splunk/var/log/splunk/Splunk_TA_otel.log" "$TEST_FOLDER/"
+    if [ "$PLATFORM"  == "windows" ]; then
+        scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -r -i ~/.orca/id_rsa "splunk@$ip_addr:/opt/splunk/var/log/splunk/Splunk_TA_otelutils.log" "$TEST_FOLDER/"
+        grep -q "INFO Otel agent stop" "$TEST_FOLDER/Splunk_TA_otelutils.log" && break
+    else
+        grep -q "INFO Otel agent stop" "$TEST_FOLDER/Splunk_TA_otel.log" && break
+    fi
+
+    ATTEMPT=$((ATTEMPT + 1))
+    sleep $DELAY
+done
+
+if [ $ATTEMPT -gt $MAX_ATTEMPTS ]; then
+    echo "Failed to see restart log m after $MAX_ATTEMPTS attempts."
+    if [ "$PLATFORM"  == "windows" ]; then
+        cat "$TEST_FOLDER/Splunk_TA_otelutils.log"
+    else
+        cat "$TEST_FOLDER/Splunk_TA_otel.log"
+    fi
+    exit 1
+fi
+
+scp -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null -r -i ~/.orca/id_rsa "splunk@$ip_addr:/opt/splunk/var/log/splunk/otel.log" "$TEST_FOLDER/"
+grep -q "Starting otel agent" "$TEST_FOLDER/splunk/Splunk_TA_otel.log"
+grep -q "Everything is ready" "$TEST_FOLDER/splunk/otel.log"
+(grep -qi "ERROR" "$TEST_FOLDER/splunk/Splunk_TA_otel.log" && exit 1 ) || true
+(grep -qi "ERROR" "$TEST_FOLDER/splunk/otel.log" && exit 1 ) || true
+
+# Ensure version is as expected
+actual_version="$(grep "Version" "$TEST_FOLDER/splunk/otel.log" | head -1 | awk -F 'Version": "' '{print $2}' | awk -F '", "' '{print $1}')"
+echo "actual version: $actual_version"
+[[ "$actual_version" != "v0.111.0" ]] && echo "Test failed -- invalid version" && exit 1
```

Comment:
Should we eventually have this be a more central variable for expected version? Should we document somewhere that this should be updated for every new TA release?